Fix indentation and add a new (currently failing) test

toddy15 opened the pull request:
Hi @mychidarko,

I started adding a new test for the built-in rule 'required'. However, the test currently fails on line 64.

I've made the assumption that the $errorsArray should not contain the key 'test' if I set the rule 'required' and the validation succeeds. However, the $errorsArray still contains the key 'test' from the previous (expected failing) validation.

Is this a bug in the leafsphp/form codebase or is the test assumption wrong?

mychidarko commented:
When `required` validation passes, the errorsArray should not contain the `test` key.

toddy15 commented:
Thanks for the confirmation, that's what I expected as well. So I assume that the test uncovered a bug in the form handling code?

mychidarko commented:
That's right @toddy15

mychidarko commented:
Still, push your tests anyway. Those would be used to make sure the validation behaves correctly

toddy15 commented:
Hi @mychidarko,

there are now tests for every built-in rule. I've also spotted and fixed a typo in the error messages. However, all tests which pass the validation are failing right now, because of the $errorArray still containing the old error message.

Also, the date validation seems a little odd to me - a single space (" ") counts as valid date. Is this intended?

mychidarko commented:
No that's not intended

toddy15 commented:
Now I've added a workflow for github to run the tests automatically upon pushing. I've also fixed the failing tests -- please have a look if this fix is correct from your point of view. Lastly, I've disabled the only failing test (with the date) for now, so that the github workflow succeeds.

mychidarko commented:
I'll check it out, thanks

mychidarko commented:
Wow, this is amazing @toddy15. Please let me know when its ready to be merged

mychidarko commented:
I have a couple of changes I want to make so the tests match up to the tests we have in our other repos. Minor, but just for the sake of consistency. I'll update those after the merge so you don't need to worry about them.

toddy15 commented:
I think you can just go ahead and adapt the tests.

mychidarko commented:
Right. Awesome 💪
